core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -178,6 +178,7 @@ abstract class AppBase(
   val sqlIDtoProblematic: HashMap[Long, Set[String]] = HashMap[Long, Set[String]]()
   // sqlId to sql info
   val sqlIdToInfo = new HashMap[Long, SQLExecutionInfoClass]()
+  val sqlIdToStages = new HashMap[Long, ArrayBuffer[Int]]()
```

Comment:
Can we use `HashMap[Long, Set[Int]]` ? This will have significant impact on peerformance since many parts of the code would be using `sqlIdToStages.get(sqlID).contains(StageID)`